src/esploader.ts

```diff
@@ -1,1050 +1,1055 @@
-          }
-          if (this.IS_STUB === false) {
-            // ROM code writes block to flash before ACKing
-            timeout = block_timeout;
-          }
-          await this.flash_defl_block(block, seq, timeout);
-          if (this.IS_STUB) {
-            // Stub ACKs when block is received, then writes to flash while receiving the block after it
-            timeout = block_timeout;
-          }
-        } else {
-          throw new ESPError("Yet to handle Non Compressed writes");
-        }
-        bytes_sent += block.length;
-        image = image.slice(this.FLASH_WRITE_SIZE, image.length);
-        seq++;
-        if (reportProgress) reportProgress(i, bytes_sent, totalBytes);
-      }
-      if (this.IS_STUB) {
-        await this.read_reg(this.CHIP_DETECT_MAGIC_REG_ADDR, timeout);
-      }
-      d = new Date();
-      const t = d.getTime() - t1;
-      if (compress) {
-        this.info(
-          "Wrote " +
-            uncsize +
-            " bytes (" +
-            bytes_sent +
-            " compressed) at 0x" +
-            address.toString(16) +
-            " in " +
-            t / 1000 +
-            " seconds.",
-        );
-      }
-      if (calcmd5) {
-        const res = await this.flash_md5sum(address, uncsize);
-        if (new String(res).valueOf() != new String(calcmd5).valueOf()) {
-          this.info("File  md5: " + calcmd5);
-          this.info("Flash md5: " + res);
-          throw new ESPError("MD5 of file does not match data in flash!");
-        } else {
-          this.info("Hash of data verified.");
-        }
-      }
-    }
-    this.info("Leaving...");
-
-    if (this.IS_STUB) {
-      await this.flash_begin(0, 0);
-      if (compress) {
-        await this.flash_defl_finish();
-      } else {
-        await this.flash_finish();
-      }
-    }
-  }
-
-  async flash_id() {
-    this.debug("flash_id");
-    const flashid = await this.read_flash_id();
-    this.info("Manufacturer: " + (flashid & 0xff).toString(16));
-    const flid_lowbyte = (flashid >> 16) & 0xff;
-    this.info("Device: " + ((flashid >> 8) & 0xff).toString(16) + flid_lowbyte.toString(16));
-    this.info("Detected flash size: " + this.DETECTED_FLASH_SIZES[flid_lowbyte]);
-  }
-
-  async hard_reset() {
-    await this.transport.setRTS(true); // EN->LOW
-    await this._sleep(100);
-    await this.transport.setRTS(false);
-  }
-
-  async soft_reset() {
-    if (!this.IS_STUB) {
-      // "run user code" is as close to a soft reset as we can do
-      await this.flash_begin(0, 0);
-      await this.flash_finish(false);
-    } else if (this.chip.CHIP_NAME != "ESP8266") {
-      throw new ESPError("Soft resetting is currently only supported on ESP8266");
-    } else {
-      // running user code from stub loader requires some hacks
-      // in the stub loader
-      await this.command(this.ESP_RUN_USER_CODE, undefined, undefined, false);
-    }
-  }
-}
+import { ESPError } from "./error";

```

Comment:
You've changed the line ending on these file, causing each line to be marked as changed. Please revert that.

Reply:
These should be reverted now.